tests/test_readwrite.py

```diff
@@ -27,121 +27,101 @@ def factory_test_graph(ir: nir.NIRGraph):
 
 
 def test_simple():
-    ir = nir.NIRGraph(nodes=[nir.Affine(weight=[1, 2, 3], bias=4)], edges=[(0, 0)])
+    ir = nir.NIRGraph(
+        nodes={"a": nir.Affine(weight=[1, 2, 3], bias=4)}, edges=[("a", "b")]
```

Comment:
There is no node "b", so this test should be failing!

Reply:
Hehe, true!